infrastructure/simulation-tool/src/operations/transfer.ts

```diff
@@ -0,0 +1,101 @@
+import { BigNumber, Wallet as EthersWallet, constants } from 'ethers';
+import { Wallet as RollupWallet, Transaction, utils } from '@rsksmart/rif-rollup-js-sdk/';
+import config from '../utils/config.utils';
+import { getRandomBigNumber } from '../utils/number.utils';
+import { TransactionReceipt } from '@rsksmart/rif-rollup-js-sdk/build/types';
+import { sleep } from '@rsksmart/rif-rollup-js-sdk/build/utils';
+import { chooseRandomWallet } from '../utils/wallet.utils';
+
+type PreparedTransfer = Omit<Parameters<RollupWallet['syncTransfer']>[number], 'amount'> & {
+    amount: BigNumber;
+    from: RollupWallet;
+};
+
+type TransferResult = {
+    opL2Receipt: TransactionReceipt;
+    verifierReceipt: TransactionReceipt;
+};
+
+const prepareTransfer = (l2sender: RollupWallet, recipient: EthersWallet): PreparedTransfer => {
```

Comment:
TBH I preferred to pass the whole wallet to have it nicely typed. Otherwise, it's just a string with no useful type check.

Reply:
actually I'll just do what you say. It will make it simpler.

Reply:
changed

Reply:
@jurajpiar it looks like this one is still valid

Reply:
I don't know what happened there, but I guess I was looking at a different thing? I don't know. Furthermore, I feel like I had a stroke.
Anyway, the sender has to be there, because we want to have complete information about the transaction. This is so that we can simply iterate over these prepared transactions (be it transfer, deposit, or whatever), instead of iterating over senders & transactions, when we want to execute.

Reply:
Yes, but my question is: do we need a wallet for the receiver, or an address would be enough?